tests/drivers/gpio/gpio_get_direction/src/main.c

```diff
@@ -17,6 +17,11 @@ static void test_disconnect(void)
 	zassert_not_null(port, "device " TEST_DEV " not found");
```

Comment:
```
port = device_get_binding(TEST_DEV);
zassert_not_null(port, "device " TEST_DEV " not found");
```

Seems to be recurring, may I suggest (you can use zassert if you don't want to wait for the assume API, but then you should do the assert in the before function instead so it doesn't blow up your main thread):

```
struct gpio_get_direction_fixture {
  const struct device *port;
};

static void *gpio_get_direction_setup(void)
{
  static struct gpio_get_direction_fixture fixture;

  fixture.port = device_get_binding(TEST_DEV);
  zassume_not_null(port, "device " TEST_DEV " not found");

  return &fixture;
}

// Create the test suite and add the setup function
ZTEST_SUITE(gpio_get_direction, NULL, gpio_get_direction_setup, NULL, NULL, NULL);

// The _F tells it to use the fixture provided by the setup function
ZTEST_F(gpio_get_direction, test_disconnect)
{
  ...
  // Use fixture->port here
}
```

Reply:
I'll change it when the zassume work is done. I can see that would be helpful here.